app/src/main/java/com/huji/foodtricks/buddies/EventsTabsActivity.java

```diff
@@ -264,4 +271,40 @@ public boolean onCreateOptionsMenu(Menu menu) {
         inflater.inflate(R.menu.events_tabs_menu, menu);
         return true;
     }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.signoutButton:
+                signOut();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+
+    private void signOut() {
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build();
+
+        GoogleSignInClient mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
+
+        FirebaseAuth mAuth = FirebaseAuth.getInstance();
+
+        // Firebase sign out
+        mAuth.signOut();
+
+        // Google sign out
+        mGoogleSignInClient.signOut().addOnCompleteListener(this, new OnCompleteListener<Void>() {
+            @Override
+            public void onComplete(@NonNull Task<Void> task) {
+                Intent signoutIntent = new Intent(EventsTabsActivity.this, SignInActivity.class);
```

Comment:
To remove login/signout logic from this activity, I think it's better to use finish(), to send the user back to the previous activity (sign in page), and not accidentally create redundant activities on the stack.
You can use StartActivityForResult on the sign-in activity, and send "SIGN OUT" result from this activity,
and handle OnActivityResult for this specific result (just move the sign out method to the sign in activity).

You can look at the CreateGroupActivity methods onCreateGroupPressed() and onBackPressed().